src/client/actions.rs

```diff
@@ -13,26 +13,56 @@ use crate::{
 use anyhow::Result;
 use ed25519_dalek::VerifyingKey;
 use futures::future::join_all;
+use bon::bon;
+use std::time::{Duration, SystemTime};
 
 use super::{DeletionEvent, Event, SetEvent};
 
 pub struct Actions {
     pub config: Configuration,
 }
 
+pub enum Expiry {
+    ExpiresAt(u64),
+    Ttl(Duration),
+}
+
+#[bon]
 impl Actions {
     pub fn new(config: Configuration) -> Actions {
         Actions { config }
     }
 
-    async fn set_internal(&self, name: Name, value: Value, expires_at: Option<u64>) -> Result<()> {
+    #[builder]
+    pub async fn set(&self, name: Name, value: Value, expiry: Option<Expiry>, priority: Option<u64>) -> Result<()> {
```

Comment:
Maybe in the future we can set up name and value using `#[builder(start_fn)] ` http://bon-rs.com/guide/positional-members

This way it might be a few less lines when calling the function
```rust
actions.set(name, value).expiry(...).call()
```